services/report.py

```diff
@@ -224,3 +224,41 @@ def build_files(commit_report: CommitReport) -> dict[str, ReportFileSummary]:
         )
         for file in report_details.files_array
     }
+
+
+def files_belonging_to_flags(commit_report: Report, flags: List[str]) -> List[str]:
+    sessions_for_specific_flags = calculate_sessions_with_specific_flags(
+        commit_report=commit_report, flags=flags
+    )
+    session_ids = list(sessions_for_specific_flags.keys())
+    files_in_specific_sessions = calculate_files_in_sessions(
+        commit_report=commit_report, session_ids=session_ids
+    )
+    return files_in_specific_sessions
+
+
+def calculate_sessions_with_specific_flags(
```

Comment:
Not a huge fan of including things like `calculate` or `compute` in function names since that's basically the function of a computer.  Maybe just `sessions_with_flags`?

Reply:
Yeah that's fine, I remember seeing somewhere this as a "good naming convention" as this is describing a verb instead of a noun, where nouns should be more like @propertys (which I can change to too)

Reply:
Hmm, interesting.  I'd say if it's mutating state somewhere then a verb definitely makes sense.  If it's just returning data then maybe less important to make that distinction?

Reply:
Yeaaah makes sense to me, I can adjust this accordingly 👌